python/triton/runtime/jit.py

```diff
@@ -277,50 +277,61 @@ def has_default(self):
 
 
 dtype2str = {}
+specialize_impl_cache = []
 
 
-def specialize_impl(arg, specialize_extra, is_const=False, specialize_value=True, align=True):
+def create_specialize_impl():
+    if len(specialize_impl_cache) > 0:
```

Comment:
```suggestion
    if specialize_impl_cache:
```

Might offer a slight improvement

Reply:
@Jokeren This isn't on the hot codepath, but I'll commit your suggestion anyway because it makes the code simpler